core/src/environment/mod.rs

```diff
@@ -72,7 +72,9 @@ pub struct Environment {
 
     /// The [`EVM`] that is used as an execution environment and database for
     /// calls and transactions.
-    pub(crate) db: ArbiterDB,
+    pub(crate) tip_db: ArbiterDB,
```

Comment:
Whats the intuition of calling it tip? Like it's the front of a queue? Or top of a stack?